libcxx/include/__exception/exception_ptr.h

```diff
@@ -51,11 +69,24 @@ class _LIBCPP_EXPORTED_FROM_ABI exception_ptr {
 template <class _Ep>
 _LIBCPP_HIDE_FROM_ABI exception_ptr make_exception_ptr(_Ep __e) _NOEXCEPT {
 #  ifndef _LIBCPP_HAS_NO_EXCEPTIONS
+#    if defined(LIBCXX_BUILDING_LIBCXXABI)
```

Comment:
This should be guarded by an availability macro instead.

Reply:
Should it be a macro or a runtime availability check instead?

Reply:
I've added support for `libcxxrt` as well and made it a runtime check

Reply:
The `ifdef` became a bit too big to copy-paste it, so i moved it into `__config`